### PR TITLE
fmt 9 ostream fix

### PR DESCRIPTION
**Pull request by longlene**
include fmt/core.h before check FMT_VERSION
define deprecation macro as a workaround as struct defined in anonymous namespace

### CODE REVIEW

File: include/seastar/testing/perf_tests.hh [hunk 1]
```diff
@@ -24,6 +24,7 @@
 #include <atomic>
 #include <memory>
 
+#define FMT_DEPRECATED_OSTREAM
```
Comment (xemul) [inline include/seastar/testing/perf_tests.hh:27]: Please, explain this change in more details

Comment (longlene) [reply include/seastar/testing/perf_tests.hh:27]: @xemul https://github.com/fmtlib/fmt/releases/tag/9.0.0
According to the release notes of fmt 9.0.0, automatic std::ostream insertion operator (operator<<) discovery has been disabled and it's also hard to define formatter specializations for the anonymous struct.

So it's just a workaround.

Comment (avikivity) [reply include/seastar/testing/perf_tests.hh:27]: It's better to fix it for real, since deprecated features eventually get removed.

Comment (longlene) [reply include/seastar/testing/perf_tests.hh:27]: @avikivity Right, if we can move the struct duration to a named namespace, then we can define the formatter specialization.
Anyway, I'll take a try

**Comment by BenPope** [comment]
I made a similar change: https://github.com/scylladb/seastar/pull/1188

File: tests/perf/perf_tests.cc [hunk 2]
```diff
@@ -160,7 +167,6 @@ struct result {
     double inst = 0.;
 };
 
-namespace {
```
Comment (xemul) [inline tests/perf/perf_tests.cc:163]: Please, re-split the series. First patch is to de-anonymize the duration class. Second patch to add formatters for the missing pieces.

Comment (longlene) [reply tests/perf/perf_tests.cc:163]: @xemul Please take a look again

**Comment by xemul** [comment]
@longlene , looks like 2nd patch -- with the formatter itself -- is missing

**Comment by longlene** [comment]
@xemul I'll create another pr, is it ok?

**Comment by xemul** [comment]
I'd prefer having one PR with two patches -- de-anonymization and formatter. Be it in this PR or in the new one -- I'm fine with either.

**Comment by longlene** [comment]
@xemul Anyway, I have no idea how to create "one PR with two patches"?

**Comment by xemul** [comment]
@longlene , yes -- two commits one PR

**Comment by longlene** [comment]
ok, I'll do it later, thank you!

**Pull request closed without merging**